### PR TITLE
[COST-3083] - enable volume labels on OCP cost endpoint

### PR DESCRIPTION
**Pull request by myersCody**
## Jira Ticket

[COST-3083](https://issues.redhat.com/browse/COST-3083)

## Description

This change will add a new field to our OCP usage postgress table and populate with the combined volume + pod labels. We then also switch which field the provider_map is using making pod AND volume labels accessible from the OpenShift cost API.

## Testing

1. Checkout Branch
2. Let migration run
4. hit the following endpoint with volume and pod label group bys `/api/cost-management/v1/reports/openshift/costs/?group_by[tag:{volume_labels/pod_labels}]=*`

5. Hit the following endpoint:
```
http://localhost:8000/api/cost-management/v1/reports/openshift/costs/?group_by[tag:app]=*
```
6. Check the logs:
```
koku_server         | [2023-09-18 18:44:44,587] WARNING None 26 DEVELOPMENT is Enabled. Bypassing access lookup for user: {'identity': {'account_number': '10001', 'org_id': '1234567', 'type': 'User', 'user': {'username': 'user_dev', 'email': 'user_dev@foo.com', 'is_org_admin': 'True', 'access': {}}}, 'entitlements': {'cost_management': {'is_entitled': 'True'}}}
koku_server         | [2023-09-18 18:44:44,610] WARNING None 26 Returning default value for feature: cost-management.backend.override_customer_group_by_limit
koku_server         | [2023-09-18 18:44:44,610] WARNING None 26 Attempted to get feature_flag cost-management.backend.override_customer_group_by_limit, but client wasn't initialized!
koku_server         | [2023-09-18 18:44:44,638] WARNING None 26 Returning default value for feature: cost-management.backend.feature-cost-3083-all-labels
koku_server         | [2023-09-18 18:44:44,638] WARNING None 26 Attempted to get feature_flag cost-management.backend.feature-cost-3083-all-labels, but client wasn't initialized!
koku_server         | [2023-09-18 18:44:44,639] WARNING None 26 ('tag:app',) for costs has no entry in views. Using the default.
koku_server         | [2023-09-18 18:44:44,651] WARNING None 26 Exchange rates dictionary is not populated resulting in 'NoneType' object has no attribute 'currency_exchange_dictionary'.
```
Notice how substring `Returning default value for feature: cost-management.backend.feature-cost-3083-all-labels` is in the return.

7. Add a limit to the url:
```
http://localhost:8000/api/cost-management/v1/reports/openshift/costs/?group_by[tag:app]=*&limit=100
```
Check the logs:
```
koku_server         | [2023-09-18 18:45:31,652] WARNING None 26 DEVELOPMENT is Enabled. Bypassing access lookup for user: {'identity': {'account_number': '10001', 'org_id': '1234567', 'type': 'User', 'user': {'username': 'user_dev', 'email': 'user_dev@foo.com', 'is_org_admin': 'True', 'access': {}}}, 'entitlements': {'cost_management': {'is_entitled': 'True'}}}
koku_server         | [2023-09-18 18:45:31,752] WARNING None 26 Returning default value for feature: cost-management.backend.override_customer_group_by_limit
koku_server         | [2023-09-18 18:45:31,753] WARNING None 26 Attempted to get feature_flag cost-management.backend.override_customer_group_by_limit, but client wasn't initialized!
koku_server         | [2023-09-18 18:45:31,773] WARNING None 26 ('tag:app',) for costs has no entry in views. Using the default.
koku_server         | [2023-09-18 18:45:31,783] WARNING None 26 Exchange rates dictionary is not populated resulting in 'NoneType' object has no attribute 'currency_exchange_dictionary'.
```
Notice how `Returning default value for feature: cost-management.backend.feature-cost-3083-all-labels` does not appear in the logs now.

8. Log into unleash:
```
http://localhost:4242/
```
9. Add `org1234567` the `cost-management.backend.feature-cost-3083-all-labels` flag.

## Notes

...

### CODE REVIEW

**Comment by myersCody** [comment]
Smoke Test Failures before unleash:
- https://ci.ext.devshift.net/job/project-koku-koku-pipeline-pr-check-main/489/

**Comment by myersCody** [comment]
Smokes after unleash check:
- https://ci.ext.devshift.net/job/project-koku-koku-pipeline-pr-check-main/496/

File: koku/api/report/queries.py [hunk 1]
```diff
@@ -118,6 +128,9 @@ def __init__(self, parameters):
         self._offset = parameters.get_filter("offset", default=0)
         self.query_delta = {"value": None, "percent": None}
         self.query_exclusions = None
+        self._tag_column = check_unleash_for_tag_column_cost_3038(
+            self._mapper.tag_column, parameters.tenant.schema_name
+        )
```
Comment (maskarb) [inline koku/api/report/queries.py:133]: I really wish that this check did not live here. I understand we don't have the parameters in the ProviderMap, but that's where this thing should live. The `mapper` is responsible for the tag_column, not the query handler.

I think if we want the ability to control mapper values via unleash, then we need to update all the mappers to accept the schema. Then we can define flags in specific mappers when we need to. All of this work here is specific to OCP, so it doesn't make sense to me to add it to the base report query handler which is used by all providers.

**Review by maskarb — DISMISSED** [review]

**Comment by myersCody** [comment]
While working on fixing the smoke tests locally, I was getting some bizarre results compared to what I was seeing locally. Turns out while developing locally, I used the migration to populate the all_labels field. 

The good news is I confirmed our migration will correlate populate the all_labels field for the cost model rows. Bad news is I discovered we are not populating that column on our initial ingestion after that migration is ran 😓 

I have [fixed the cost model sql](https://github.com/project-koku/koku/pull/4687/commits/2486b0c0b9194923b108bec64b0f708d3ab6e4f7) to also populate the all_labels any time the volume_labels or pod_labels fields are being populated.

Further Testing Instructions:
1. Switch to this branch
2. Get a Fresh DB
3. create test customer data
4. Query the database:
```
select * from reporting_ocpusagelineitem_daily_summary where usage_start = '2023-09-20' and all_labels ? 'app' and cost_model_rate_type IS NOT NULL;
```
^ This sql used to return None before I updated the cost_model sql files.

5. We can also take it a step further now. Lets see if there is any row in the database where pod_labels does not equal all labels:
```
select * from reporting_ocpusagelineitem_daily_summary where pod_labels != all_labels;
(0 rows)
```

6. We can do the same with the volume_labels now.
```
postgres=# select * from reporting_ocpusagelineitem_daily_summary where volume_labels != all_labels;
(0 rows)
```
7. Other checks I did:
```
postgres=# select count(*) from reporting_ocpusagelineitem_daily_summary where pod_labels is not null and all_labels is not null and cost_model_rate_type is not null and usage_start > '2023-09-01';
 count
-------
   126
(1 row)

postgres=# select count(*) from reporting_ocpusagelineitem_daily_summary where pod_labels = all_labels and cost_model_rate_type is not null and usage_start > '2023-09-01';
 count
-------
   126
(1 row)

postgres=# select count(*) from reporting_ocpusagelineitem_daily_summary where pod_labels !=  all_labels and cost_model_rate_type is not null and usage_start > '2023-09-01';
 count
-------
     0
(1 row)

select count(*) from reporting_ocpusagelineitem_daily_summary where pod_labels !=  all_labels and monthly_cost_type is not null and usage_start > '2023-09-01';
```

**Review by maskarb — APPROVED** [review]

**Comment by myersCody** [comment]
OCP Smoke tests:
- https://ci.ext.devshift.net/job/project-koku-koku-pipeline-pr-check-main/602/

One failure that I believe is unrelated.

**Review by lcouzens — APPROVED** [review]

**Pull request merged**